src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -194,15 +197,32 @@ define(function (require, exports, module) {
             });
         }
 
-        add(root, function(err) {
-            if(err) {
-                return callback(err);
+        // Check if path is a directory or not
+        fs.stat(path, function(err, stats) {
+            if (stats.type === "DIRECTORY") {
+                add(path, function(err) {
+                    if(err) {
+                        return callback(err);
+                    }
+                    // Prepare folder for download
+                    var compressed = jszip.generate({type: 'arraybuffer'});
+                    var blob = new Blob([compressed], {type: "application/zip"});
+                    saveAs(blob, "project.zip");
+                    callback();
+                });
+            } else {
+                // Prepare file for download
+                fs.readFile(path, {encoding: null}, function(err, data){
```

Comment:
I don't think it makes sense to do this inside a function called `archive`.  If I pass you a `path` to `archive`, you should always produce a `.zip`.  Instead of this, you should check at your call site to see if this is a file or folder, and only call `archive` in the latter case, otherwise do this code.